shell/platform/android/io/flutter/embedding/android/FlutterFragment.java

```diff
@@ -76,219 +72,256 @@
-    flutterView.detachFromFlutterRenderer();
-    if (detach || retainFlutterIsolateAfterFragmentDestruction()) {
-      // Detach, but do not destroy the FlutterView if a plugin expressed interest in its
-      // FlutterNativeView.
-      flutterEngine.detachFromJni();
-    } else {
-      flutterEngine.destroy();
-    }
+    FlutterMain.ensureInitializationComplete(context.getApplicationContext(), flutterShellArgs);
   }
 
   /**
-   * The hardware back button was pressed.
+   * Creates a new FlutterEngine instance.
    *
-   * See {@link Activity#onBackPressed()}
+   * Subclasses can instantiate their own {@link FlutterEngine} by overriding
+   * {@link #onCreateFlutterEngine(Context)}.
+   *
+   * Subclasses can alter the {@link FlutterEngine} after creation by overriding
+   * {@link #onFlutterEngineCreated(FlutterEngine)}.
    */
-  public void onBackPressed() {
-    Log.d(TAG, "onBackPressed()");
-    popRoute();
+  private void createFlutterEngine() {
+    // Create a FlutterEngine to back our FlutterView.
+    flutterEngine = onCreateFlutterEngine(getActivity());
+
+    // Allow subclasses to customize FlutterEngine as desired.
+    onFlutterEngineCreated(flutterEngine);
   }
 
   /**
-   * The result of a permission request has been received.
-   *
-   * See {@link Activity#onRequestPermissionsResult(int, String[], int[])}
+   * Hook for subclasses to customize the creation of the {@code FlutterEngine}.
    *
-   * @param requestCode identifier passed with the initial permission request
-   * @param permissions permissions that were requested
-   * @param grantResults permission grants or denials
+   * By default, this method returns a standard {@link FlutterEngine} without any modification.
    */
-  public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
-    flutterEngine.getPluginRegistry().onRequestPermissionsResult(requestCode, permissions, grantResults);
+  @NonNull
+  protected FlutterEngine onCreateFlutterEngine(@NonNull Context context) {
+    Log.d(TAG, "onCreateFlutterEngine()");
+    return new FlutterEngine(context);
   }
 
   /**
-   * A new Intent was received by the {@link Activity} that currently owns this {@link Fragment}.
-   *
-   * See {@link Activity#onNewIntent(Intent)}
-   *
-   * @param intent new Intent
+   * Hook for subclasses to customize the {@link FlutterEngine} owned by this {@link FlutterFragment}
+   * after the {@link FlutterEngine} has been instantiated.
    */
-  public void onNewIntent(@NonNull Intent intent) {
-    flutterEngine.getPluginRegistry().onNewIntent(intent);
-  }
-
-  @Override
-  public void onActivityResult(int requestCode, int resultCode, Intent data) {
-    flutterEngine.getPluginRegistry().onActivityResult(requestCode, resultCode, data);
+  protected void onFlutterEngineCreated(@NonNull FlutterEngine flutterEngine) {
+    // no-op
```

Comment:
Eliminate the comment since it doesn't add much value. In its place, I'd update the method documentation to say that by default it does no customisation.